pkg/provider/aws/resources/eks.go

```diff
@@ -144,192 +142,261 @@ type (
 	}
 )
 
-// CreateEksCluster will create a cluster in the subnets provided, with the attached additional security groups
-//
-// The method will also create a fargate profile in the default namespace and a single NodeGroup.
-// The method will create all the corresponding IAM Roles necessary and attach all the execution units references to the following objects
-func CreateEksCluster(cfg *config.Application, clusterName string, vpc *Vpc, securityGroups []*SecurityGroup, units []*core.ExecutionUnit, dag *core.ResourceGraph) error {
-	references := []core.AnnotationKey{}
-	for _, u := range units {
-		references = append(references, u.Provenance())
-	}
+type EksClusterCreateParams struct {
+	Refs    []core.AnnotationKey
+	AppName string
+	Name    string
+}
 
-	appName := cfg.AppName
-	subnets := vpc.GetVpcSubnets(dag)
+func (cluster *EksCluster) Create(dag *core.ResourceGraph, params EksClusterCreateParams) error {
 
-	clusterRole := createClusterAdminRole(appName, clusterName+"-k8sAdmin", references)
-	dag.AddResource(clusterRole)
+	cluster.Name = clusterSanitizer.Apply(fmt.Sprintf("%s-%s", params.AppName, params.Name))
 
-	cluster := NewEksCluster(appName, clusterName, vpc, subnets, securityGroups, clusterRole)
-	cluster.ConstructsRef = references
-	dag.AddDependenciesReflect(cluster)
+	existingCluster := dag.GetResource(cluster.Id())
+	if existingCluster != nil {
+		graphCluster := existingCluster.(*EksCluster)
+		graphCluster.ConstructsRef = append(graphCluster.ConstructsRef, params.Refs...)
```

Comment:
do we need to deduplicate refs here?

Reply:
ah yeah good catch, will make that change